src/Bicep.Core/Configuration/ExperimentalFeaturesEnabled.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Bicep.Core.Extensions;
+using Bicep.Core.Json;
+
+namespace Bicep.Core.Configuration;
+
+public record ExperimentalFeaturesEnabled(
+    bool? SymbolicNameCodegen,
+    bool? Imports,
+    bool? ResourceTypedParamsAndOutputs,
+    bool? SourceMapping,
+    bool? ParamsFiles)
+{
+    public static ExperimentalFeaturesEnabled Bind(JsonElement element, string? configurationPath)
```

Comment:
Sorry to be difficult:
1. What benefit do we get out of this being an object vs an array? e.g. `"experimentalFeatures": ["foo", "bar"]`
1. What's the experience if the user opts-in to a feature which isn't supported by the compiler? Should we error/warning if that happens?

Reply:
Using an object lets us turn on an experimental feature by default while still giving users a way to disable it if they run into any issues. We currently support this form of transition today (e.g., if we enabled symbolic name codegen by default, a user could turn it off with `export BICEP_SYMBOLIC_NAME_CODEGEN_EXPERIMENTAL=false`).

The behavior when a user tries to enable a non-existent feature flag is a config load error, which blocks compilation. This could be converted from an error to a warning, but I think that creates a confusing situation (e.g., with `"symbolcNameCodegen": true`, we would emit a non-blocking warning and then not use symbolic names).

Reply:
I think it makes sense to start strict and wait for feedback to relax it. 

I can see a hypothetical situation where different users sharing the bicepconfig.json in a repo run different versions of bicep that have different sets of experimental features, which would force the enabled experimental features being a common denominator but I'm not terribly worried about that scenario... yet.

My preference is object as well for the same reasons. Disabling in an array requires removing the whole item from the array or comment out the item with block comments. With an object you can just change to false. (With completions that's as simple as selecting `true`, typing f and enter.)

Reply:
As far as disabling features goes - my expectation is that we won't be shipping any "experimental" features that are true by default - instead we'll just remove the property from `IFeatureProvider`.

The scenarios for the object do feel a little hypothetical - but I also don't feel strongly about it being an array vs object.